src/main/kotlin/no/nav/tilleggsstonader/kontrakter/periode/MergeOverlappende.kt

```diff
@@ -0,0 +1,41 @@
+package no.nav.tilleggsstonader.kontrakter.periode
+
+import no.nav.tilleggsstonader.kontrakter.felles.KopierPeriode
+import no.nav.tilleggsstonader.kontrakter.felles.Periode
+import java.time.LocalDate
+
+/**
+ * @param erLike skal kun slå sammen overlappende hvis overlappende OG [erLike]
+ * @param merge slår sammen 2 perioder. Trenger ikke å gjøre noe med datoene, då [periode1Snitt] allerede er snittet av de 2 periodene
+ * Vanligt usecase: {p1, p2 -> p1.copy(beløp = p1.beløp + p2.beløp)}
+ *
+ * @return alle perioder, men merger de periodene som overlapper.
+ * Dvs 1jan til 2jan og 2jan til 3jan blir
+ * 1jan til 1jan
+ * 2jan til 2jan // overlappende periode - merges, sånn at eks beløp kan slåes sammen for denne perioden
+ * 3jan til 3jan
+ */
```

Comment:
Kan vi legge inn en kommentar på hvorfor dette blir 3 perioder og ikke 1. Jeg tenkte først at disse tre er sammenhengende og da burde bli 1 periode. Eventuelt om man skulle gitt den et annet navn 🤔 

Ser i https://github.com/navikt/tilleggsstonader-sak/pull/584/files at de kan ha ulik studieprosent og derfor må vi skille de, men usikker på om jeg hadde skjønt det uten kontekst fra PR-en i sak.

Reply:
Har du forslag på hvordan det burde dokumenteres? Har nå
```text
@return alle perioder, men merger de periodene som overlapper.
...
* 2jan til 2jan // overlappende periode - merges, sånn at eks beløp kan slåes sammen for denne perioden`
```

Reply:
* @return alle perioder, men merger de periodene som overlapper.
 * Deler av periodene som ikke overlapper blir avkuttet og beholdt som egne perioder <- nytt
 * Dvs 1jan til 2jan og 2jan til 3jan blir
 * 1jan til 1jan
 * 2jan til 2jan // overlappende periode - merges, sånn at eks beløp kan slåes sammen for denne perioden
 * 3jan til 3jan